integrationtest/mapper/pojo/src/test/java/org/hibernate/search/integrationtest/mapper/pojo/mapping/definition/ScaledNumberFieldIT.java

```diff
@@ -0,0 +1,112 @@
+import org.hibernate.search.integrationtest.mapper.pojo.testsupport.util.rule.JavaBeanMappingSetupHelper;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.DocumentId;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.ScaledNumberField;
+import org.hibernate.search.util.common.SearchException;
+import org.hibernate.search.util.impl.integrationtest.common.FailureReportUtils;
+import org.hibernate.search.util.impl.integrationtest.common.rule.BackendMock;
+import org.hibernate.search.util.impl.test.SubTest;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+public class ScaledNumberFieldIT {
+
+	private static final String INDEX_NAME = "IndexName";
+
+	@Rule
+	public BackendMock backendMock = new BackendMock( "stubBackend" );
+
+	@Rule
+	public JavaBeanMappingSetupHelper setupHelper = new JavaBeanMappingSetupHelper( MethodHandles.lookup() );
+
+	@Test
+	public void validDecimalScales() {
+
+		@Indexed(index = INDEX_NAME)
+		class IndexedEntity {
+			Integer id;
+			BigDecimal scaled;
+			BigDecimal unscaled;
+			BigDecimal defaultScaled;
+
+			@DocumentId
+			public Integer getId() {
+				return id;
+			}
+
+			@ScaledNumberField(decimalScale = 7)
+			public BigDecimal getScaled() {
+				return scaled;
+			}
+
+			@ScaledNumberField(decimalScale = 0)
+			public BigDecimal getUnscaled() {
+				return unscaled;
+			}
+
+			@ScaledNumberField
+			public BigDecimal getDefaultScaled() {
+				return defaultScaled;
+			}
+		}
+
+		// Schema
+		backendMock.expectSchema( INDEX_NAME, b -> b
+				.field( "scaled", BigDecimal.class, f -> f.decimalScale( 7 ) )
+				.field( "unscaled", BigDecimal.class, f -> f.decimalScale( 0 ) )
+				.field( "defaultScaled", BigDecimal.class )
+		);
+		setupHelper.withBackendMock( backendMock ).setup( IndexedEntity.class );
+		backendMock.verifyExpectationsMet();
+	}
+
+	@Test
+	public void invalidFieldType() {
+
+		@Indexed(index = INDEX_NAME)
+		class IndexedEntity {
+			Integer id;
+			Integer notScalable;
+
+			@DocumentId
+			public Integer getId() {
+				return id;
+			}
+
+			@ScaledNumberField
+			public Integer getNotScalable() {
+				return notScalable;
+			}
+		}
+
+		SubTest.expectException(
+				() -> setupHelper.withBackendMock( backendMock ).setup( IndexedEntity.class )
+		)
+				.assertThrown()
+				.isInstanceOf( SearchException.class )
+				.hasMessageMatching( FailureReportUtils.buildFailureReportPattern()
+						.typeContext( IndexedEntity.class.getName() )
+						.pathContext( ".notScalable" )
+						.failure(
+								"This property is mapped to a scaled number field, but with a value bridge that creates a non-BigDecimal or otherwise incompatible field",
+								"bind() method returned context '",
+								"expected '" + ScaledNumberIndexFieldTypeContext.class.getName() + "'"
+						)
+						.build()
+				);
+	}
+
```

Comment:
I think that there are in the subsequent commit. Exactly the three you pointed out. I only changed the name of the third in `customBridge_invalidBinding`, since I think it would be more expressive.

Reply:
The commit is 
> HSEARCH-3487 Test bridges on ScaledNumberField annotation Fabio Massimo Ercoli 5/7/19 9:00 AM

Reply:
Right, my bad.